internal/resource/obtenantoperation/utils.go

```diff
@@ -79,3 +79,25 @@ func (m *ObTenantOperationManager) retryUpdateTenant(obj *v1alpha1.OBTenant) err
 		return m.Client.Status().Update(m.Ctx, tenant)
 	})
 }
+
+type statusMatcher func(t *v1alpha1.OBTenant) bool
```

Comment:
the input is the entire tenant, so I think it will be good to make this function type more general, maybe tenantConditionMatcher is better cause it's not limited to only status